packages/react/src/primitives/types/view.ts

```diff
@@ -7,6 +7,20 @@ type MergeProps<A, B> = A & Omit<B, keyof A>;
 
 export type ElementType = React.FC<any> | keyof JSX.IntrinsicElements;
 
+/**
+ * Convert string element type to DOMElement Type
+ * e.g. 'button' => HTMLButtonElement
+ */
+export type HTMLElementType<Element extends ElementType> =
+  Element extends keyof JSX.IntrinsicElements
+    ? JSX.IntrinsicElements[Element] extends React.DetailedHTMLProps<
```

Comment:
This could potentially be `Element extends keyof JSX.IntrinsicElements ? React.ElementRef<Element> : HTMLDivElement;` ? <- https://flow.org/en/docs/react/types/#toc-react-elementref

And can you explain why always return `HTMLDivElement` for `React.FC<any>`? 😕

Reply:
It might be more accurate to return `HTMLElement` instead of `HTMLDivElement`. Thoughts @hvergara?

Reply:
I am asking because we do have `SVGELement`, though the primitives don't need ref in large part

Reply:
yeah, good idea